packages/content/src/types.ts

```diff
@@ -0,0 +1,137 @@
+  updatedDate: Date; // formal metadata || new Date(item.modified)
+  updatedDateSource: string; // description of what was used for this attribute
+  
+  contentUrl: string; // Link to the raw content. item.url in most (but not all) item types
+  thumbnailUrl: string; // Full URL. item.thumbnail with host + path 
+  
+  license: IHubLicense; // [Future] item.licenseInfo 
+  boundary: IHubGeography; // [Future] Inline, or boolean when stored at known location /resources/boundary.json
+  
+  // Additional metadata from custom/formal elements
+  metadata?: {
+      // Unique or additional formal metadata that will be displayed in sidebar
+  };
+
+  // Hub configuration metadata
+  actionLinks?: IActionLink[]; // item.properties.links
+  actionConfigs?: Object; // item.properties.actions - enable/disable standard actions like `createWebmap` or `createStorymap`
+
+  metrics?: { // Set visibility for telemetry metrics. Nested object future-proofing, but could flatten.
+      visibility: 'private' | 'org' | 'updateGroups' | 'public'  ; // TODO: should we share with specific groups?
+  };
+
+  contentDisplay?: 'thumbnail' | 'map'; // [Future] View configuration options such as cartography, charts, table, etc.
+  source: IHubSocial; // [Future] each of these has common metadata like `title`, `thumbnail` , and `link`
+  
+  // Content specific values. Combination of relevant item.data, layer info, enrichments, configuration settings
+  // could use values instead - which is common within item.data.values
+  attributes?: {
+      // Dataset, e.g.
+      fields?: [{name:string, type:string, statistics:Array<any>}];
+      recordNumber?: number;
+      // Map, e.g.
+      layers?: [];
+      basemap?: string;
+      // Document
+      format?: 'link' | 'PDF' | 'MS Word' | 'MS Excel' | 'Text';
+      // Initiative
+      stage?: string;
+      followersNumber?: number;
+      coreTeamId?: string; 
+      contentTeamId?: string;
+      supportingTeamsNumber?: number;
+      // Event
+      sponsors?: Array<any>;
+      attendeesNumber?: number;
+      limitNumber?: number;
+      startDate?: Date;
+      endDate?: Date;
+      videoUrl?: string;
+  };
+}
+
+// Common interface to people networks: Teams, Orgs, Members
+export interface IHubSocial {
+  id: string;
+  name: string;
+  thumbnailUrl?: string;
+  region: string; // user.region - US/Europe/etc.
+  culture: string; // e.g. user.culture
+  createdDate: Date;
+  updatedDate: Date;
+  url: string;
+}
+export interface IHubOrg extends IHubSocial { }
+
+export interface IHubTeam extends IHubSocial  { }
+
+// Simple user info - more could be added/cached
+export interface IHubMember extends IHubSocial {
+  username: string; // user.username
+  thumbnailUrl?: string; // user.thumbnail
+}
+
+// title, description, and optional link to license item with more info
+export interface IHubLicense {
+  id: string; // Licenses may be Items, so provide reference for links
```

Comment:
How would we edit this? The item can only store a string in it's license field, and that's what's rendered in the Home App, so to make this work, it's really just the option to the license text, or a well-known-string (i.e. CC-BY etc) that cross-walks to preset versions of this hash. We'd have to have a UI that would present a license text editor or a well-known-license picker... we can't actually let them set these properties because where would we store it? and how would it manifest in the Home App UI?

Reply:
A component provides a UX to read/write info from the `item.licenseInfo`. If we need better license metadata, then we can add explicit parameters to `item.properties`. 

![LicensePicker](https://user-images.githubusercontent.com/1218/84096439-ebf13780-a9cf-11ea-99fa-4965dd0f9da3.gif)

Reply:
^^ the above interface is intended to search for Items of type "Content License" that contains both the explicit legal text to put into `licenseInfo` and the likely `licenseItem.properties.capabilities` noted in the type.